tests/acceptance/features/apiUpdateShare/publicShareLinkExpirationPolicy.feature

```diff
@@ -0,0 +1,247 @@
+@api
+Feature: enforce public link expiration policies
+
+  As an administrator
+  I want to enforce expiration date for public shares
+  So public shares have default expiration if the expiration date is not changed
+
+  Background:
+    Given user "Alice" has been created with default attributes and skeleton files
+
+  Scenario Outline: user tries to create a public share without expiration date and password when "days maximum until link expires if password is not set" is enabled
+    Given using OCS API version "<ocs-api-version>"
+    Given the administrator has enabled the days until link expires if password is not set public link password policy
+    When user "Alice" creates a public link share of folder "PARENT" using the sharing API
+    Then the HTTP status code should be "<http-status>"
+    Then the OCS status code should be "<ocs-status>"
+    And the value of the item "//message" in the response should match "/An expiration date is required./"
+    Examples:
+      | ocs-api-version | ocs-status | http-status |
+      | 1               | 403        | 200         |
+      | 2               | 403        | 403         |
+
+  Scenario Outline: user creates a public share without expiration date when "days maximum until link expires if password is not set" is enabled
+    Given using OCS API version "<ocs-api-version>"
+    Given the administrator has enabled the days until link expires if password is not set public link password policy
+    When user "Alice" creates a public link share using the sharing API with settings
+      | password | abcdefgh |
+      | path     | PARENT   |
+    Then the HTTP status code should be "<http-status>"
+    And the OCS status code should be "<ocs-status>"
+    And the public should be able to download file "parent.txt" from inside the last public shared folder using the new public WebDAV API with password "abcdefgh" and the content should be "ownCloud test text file parent" plus end-of-line
```

Comment:
We already have a step like this in core, just without `plus end-of-line` at the end.
It is a bit unfortunate that the step words are so long! But that needs a refactoring in core so that the first part of the step exists by itself. Let's do that in a future PR.

Reply:
Issue https://github.com/owncloud/core/issues/37671 to do the refactoring.